### PR TITLE
[Numpy] nonzero

### PR DESCRIPTION
**Pull request by tingying2020**
## Description ##
numpy operator nonzero

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)

### Changes ###
- Implement operator nonzero.

## Comments ##
The return ndarray is same as numpy.transpose(numpy.nonzero(a)).

@haojin2

### CODE REVIEW

File: src/operator/numpy/np_nonzero_op.cc [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+namespace mxnet {
+namespace op {
+
+bool NonzeroType(const nnvm::NodeAttrs& attrs,
+                 std::vector<int> *in_attrs,
+                 std::vector<int> *out_attrs) {
+  CHECK_EQ(in_attrs->size(), 1);
+  CHECK_EQ(out_attrs->size(), 1);
+  //Output must be int64.
+  TYPE_ASSIGN_CHECK(*out_attrs, 0, 6);
```
Comment (reminisce) [inline src/operator/numpy/np_nonzero_op.cc]: Use `kFloat64` instead of hard-code number.

File: src/operator/numpy/np_nonzero_op.cu [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+#include "np_nonzero_op-inl.h"
+#include <cub/cub.cuh>
+
+namespace mxnet {
+namespace op {
+
+#define MAXDIM 10
+
+inline void NonzeroForwardGPU(const nnvm::NodeAttrs& attrs,
```
Comment (reminisce) [inline src/operator/numpy/np_nonzero_op.cu]: Please delete `inline`.

File: src/operator/numpy/np_nonzero_op-inl.h [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+#include <string>
+#include <utility>
+#include <algorithm>
+#include "../operator_common.h"
+#include "../mxnet_op.h"
+#include "../tensor/init_op.h"
+#include "../mshadow_op.h"
+#include "../elemwise_op_common.h"
+
+#define NONZERO_NDIM_SWITCH(NDim, ndim, ...)       \
```
Comment (reminisce) [inline src/operator/numpy/np_nonzero_op-inl.h]: Why do we need 10 ndims?

File: src/operator/numpy/np_nonzero_op-inl.h [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    }
+  }
+};
+
+struct NonzeroForwardKernel {
+  template<int ndim>
+  MSHADOW_XINLINE static void Map(int i,
+                                  int64_t* out,
+                                  const int32_t* idx,
+                                  const mshadow::Shape<ndim>& shape) {
```
Comment (reminisce) [inline src/operator/numpy/np_nonzero_op-inl.h]: Should pass by value for GPUs.

File: src/operator/numpy/np_nonzero_op.cu (hunk 2 above)
Comment (tingying2020) [reply src/operator/numpy/np_nonzero_op.cu]: Fixed, Thanks.

File: src/operator/numpy/np_nonzero_op.cc (hunk 1 above)
Comment (tingying2020) [reply src/operator/numpy/np_nonzero_op.cc]: Fixed, Thanks.

File: src/operator/numpy/np_nonzero_op-inl.h (hunk 3 above)
Comment (tingying2020) [reply src/operator/numpy/np_nonzero_op-inl.h]: Change to 5 ndims.

File: src/operator/numpy/np_nonzero_op-inl.h (hunk 4 above)
Comment (tingying2020) [reply src/operator/numpy/np_nonzero_op-inl.h]: Fixed in new commit

**Pull request closed without merging**